src/components/metadata/templates/LocalContextsField.js

```diff
@@ -0,0 +1,241 @@
+
+            let newAVUs = avu.avus || [];
+
+            const currentLabels = newAVUs
+                .filter(
+                    (childAVU) => childAVU.attr === LocalContextsAttrs.RIGHTS_ID
+                )
+                ?.map((childAVU) => childAVU.value);
+
+            const missingLabels = newLabels.filter(
+                (label) => !currentLabels.includes(label)
+            );
+
+            const extraLabels = currentLabels.filter(
+                (label) => !newLabels.includes(label)
+            );
+
+            const requiresUpdate =
+                avu.value !== newValue ||
+                missingLabels.length > 0 ||
+                extraLabels.length > 0;
+
+            if (requiresUpdate) {
+                newAVUs = newAVUs.filter(
+                    (childAVU) =>
+                        childAVU.attr !== LocalContextsAttrs.RIGHTS_URI &&
+                        childAVU.attr !== LocalContextsAttrs.RIGHTS_ID_SCHEME &&
+                        childAVU.attr !== LocalContextsAttrs.SCHEME_URI &&
+                        (childAVU.attr !== LocalContextsAttrs.RIGHTS_ID ||
+                            newLabels.includes(childAVU.value))
+                );
+
+                newAVUs = [
+                    ...newAVUs,
+                    rightsURIAVU,
+                    schemeURIAVU,
+                    rightsIDSchemeURIAVU,
+                    ...missingLabels.map((label) => ({
+                        attr: LocalContextsAttrs.RIGHTS_ID,
+                        value: label,
+                        unit: "",
+                    })),
+                ];
+
+                onUpdate({ ...avu, value: newValue, avus: newAVUs });
+            }
+        },
+        onError: (error) => {
+            setProjectHubError(
+                <ErrorTypographyWithDialog
+                    errorMessage={t("localContextsHubError")}
+                    errorObject={error}
+                />
+            );
+        },
+    });
+
+    const updateProjectHubURI = (uri) => {
+        setProjectHubURI(uri);
+        setProjectHubError(null);
+
+        let newAVUs = avu.avus || [];
+
+        if (rightsURIAVU.value !== uri) {
+            const newRightsURIAVU = {
+                attr: LocalContextsAttrs.RIGHTS_URI,
+                value: uri,
+                unit: "",
+            };
+            setRightsURIAVU(newRightsURIAVU);
+
+            newAVUs = newAVUs.filter(
+                (childAVU) =>
+                    childAVU.attr !== LocalContextsAttrs.RIGHTS_URI &&
+                    childAVU.attr !== LocalContextsAttrs.RIGHTS_ID_SCHEME &&
+                    childAVU.attr !== LocalContextsAttrs.SCHEME_URI
+            );
+
+            newAVUs = [
+                ...newAVUs,
+                newRightsURIAVU,
+                schemeURIAVU,
+                rightsIDSchemeURIAVU,
+            ];
+
+            onUpdate({ ...avu, avus: newAVUs });
+        }
+    };
```

Comment:
This function is called whenever the user updates the Local Contexts Hub project URI in the form field, and if the URI does not match the `rightsURI` child AVU's value, then it is updated, and the other child AVUs required by DataCite v4.2 are also included.